File: androidapp/app/src/main/java/com/example/lanweather/data/Period.kt
package com.example.lanweather.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "Periods")
data class Period(
    @PrimaryKey @ColumnInfo(name = "startTime")
    val startTime: String?,
    @ColumnInfo(name = "number")
    val number: Int?,
    @ColumnInfo(name = "name")
    val name: String?,
    @ColumnInfo(name = "endTime")
    val endTime: String?,
    @ColumnInfo(name = "isDaytime")
    val isDayTime: Boolean?,
    @ColumnInfo(name = "temperature")
    val temperature: Int,
    @ColumnInfo(name = "temperatureUnit")
    val temperatureUnit: Char?,
    @ColumnInfo(name = "temperatureTrend")
    val temperatureTrend: String?,
    @ColumnInfo(name = "windSpeed")
    val windSpeed: String?,
    @ColumnInfo(name = "windDirection")
    val windDirection: Char?,
    @ColumnInfo(name = "icon")
    val icon: String?,
    @ColumnInfo(name = "shortForecast")
    val shortForecast: String?,
    @ColumnInfo(name = "detailedForecast")
    val detailedForecast: String?
) {
}

File: flutterapp/lib/models/period.dart
import 'package:json_annotation/json_annotation.dart';

part 'period.g.dart';

@JsonSerializable()
class Period {
  final String startTime;
  final int? number;
  final String? name;
  final String? endTime;
  final bool? isDaytime;
  final int? temperature;
  final String? temperatureUnit; // this was a Char in Kotlin
  final String? temperatureTrend;
  final String? windSpeed;
  final String? windDirection;
  final String? icon;
  final String? shortForecast;
  final String? detailedForecast;

  Period(
    this.startTime,
    this.number,
    this.name,
    this.endTime,
    this.isDaytime,
    this.temperature,
    this.temperatureUnit,
    this.temperatureTrend,
    this.windSpeed,
    this.windDirection,
    this.icon,
    this.shortForecast,
    this.detailedForecast,
  );

  factory Period.fromJson(Map<String, dynamic> json) => _$PeriodFromJson(json);

  Map<String, dynamic> toJson() => _$PeriodToJson(this);
}
